Description: fixes for docs that could not be auto merged
 .
 hedgehog (2.1.0~b2-1) trusty; urgency=medium
 .
   * First public packaged release
Author: John Dickinson <jad@sinodun.com>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- /dev/null
+++ hedgehog-2.1.0~b2/docs/src/Hedgehog_Installation_Guide.xml
@@ -0,0 +1,268 @@
+
+<h1><ac:structured-macro ac:name="toc" /></h1>
+<p>The following instructions describe how to install Hedgehog 2.0.0rc2 on Ubuntu 14.04 Server and set up imports of data.</p>
+<p>&nbsp;</p><ac:structured-macro ac:name="numberedheadings"><ac:rich-text-body>
+<h2>Requirements</h2>
+<h3>Platform support</h3>
+<p>Version 2.0.0rc2 of Hedgehog is currently only supported on Ubuntu 14.04.2 LTS Server.</p>
+<h3>Database support</h3>
+<p>Hedgehog has been tested with PostgreSQL 9.3.X. The installation of this of this is not covered in detail since it may or may not be co-located with the other hedgehog components. If required, it can be installed using</p><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[sudo apt-get install postgresql]]></ac:plain-text-body></ac:structured-macro><ac:structured-macro ac:name="warning"><ac:rich-text-body>
+<p>The Hedgehog servers and PostgreSQL must both be configured to use UTC!</p></ac:rich-text-body></ac:structured-macro>
+<h3>Hedgehog source code</h3>
+<p>Download the 2.0.0rc2 release from github: <a href="https://github.com/dns-stats/hedgehog/archive/2.0.0rc2.tar.gz">https://github.com/dns-stats/hedgehog/archive/2.0.0rc2.tar.gz</a></p>
+<p>(If you downloaded the above URL with wget the tarball will be called 2.0.0rc2.tar.gz)</p>
+<h2>Hedgehog components</h2>
+<p><span>Hedgehog comprises 2 components which can be run on the same or different machines:</span></p>
+<ul>
+<li>Data Manager
+<ul>
+<li>Scripts for database creation and management</li>
+<li>XML/DAT processing</li></ul></li>
+<li>Web front-end</li></ul>
+<p>Both components can be configured to connect to a remote database server, so the database does not need to be co-located with either component.</p>
+<h2>Hedgehog dependencies</h2><ac:structured-macro ac:name="info"><ac:rich-text-body>
+<p>It is necessary to only select the dependencies require for the specific components being installed.</p></ac:rich-text-body></ac:structured-macro><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[# common dependencies
+sudo apt-get install autoconf libtool make postgresql-client libyaml-tiny-perl
+# data manager dependencies
+sudo apt-get install g++ libboost-program-options-dev \
+                     libboost-system-dev libboost-filesystem-dev libpqxx-dev
+# web front-end dependencies
+sudo add-apt-repository ppa:opencpu/rapache
+sudo apt-get update
+sudo apt-get install libapache2-mod-r-base r-cran-ggplot2 r-cran-dbi \
+                     r-cran-cairodevice r-cran-reshape r-cran-digest \
+                     r-base r-base-core libcairo2-dev libxt-dev]]></ac:plain-text-body></ac:structured-macro>
+<h2>Build Hedgehog</h2>
+<p>&nbsp;</p><ac:structured-macro ac:name="info"><ac:rich-text-body>
+<ul>
+<li>By default, both components are built.&nbsp;</li>
+<li>Use the --disable-web and/or --disable-data-manager configure options to select which components are actually built.</li>
+<li>All installs get the hedgehog.yaml configuration file.</li></ul></ac:rich-text-body></ac:structured-macro>
+<p><strong style="line-height: 1.4285715;">Building Hedgehog</strong></p><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[tar -xzf hedgehog-2.0.0rc2.tar.gz
+cd hedgehog-2.0.0rc2
+./autogen.sh
+mkdir build
+cd build
+../configure [--prefix=] [--disable-web] [--disable-data-manager]
+make
+sudo make install]]></ac:plain-text-body></ac:structured-macro>
+<p>&nbsp;</p>
+<h2>Create a database</h2>
+<p><span>Ask your DBA to create the necessary database. This is a script to help them. It create the DATABASE, USERS and ROLES needed to run hedgehog (using default values), and can optionally accept a user specified database name and read/write user names.</span></p><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[<prefix>/libexec/hedgehog/database_create]]></ac:plain-text-body></ac:structured-macro>
+<p>&nbsp;</p>
+<h2>Create system users and groups</h2>
+<p>Two system users are required for Hedgehog, depending on what components are installed:</p>
+<table>
+<tbody>
+<tr>
+<th>User</th>
+<th>Default</th>
+<th colspan="1">Component</th>
+<th>Note</th></tr>
+<tr>
+<td>Database owner</td>
+<td>hedgehog</td>
+<td colspan="1">
+<p>Data Manager</p></td>
+<td>
+<p><span>This is the user that will own the database created by</span></p>
+<p><span>&nbsp;</span>hedgehog and the top level datafile directories.</p></td></tr>
+<tr>
+<td>Database read user</td>
+<td>www-data</td>
+<td colspan="1">
+<p>Web front-end</p></td>
+<td>
+<p>This defaults to the same as the default apache user</p></td></tr></tbody></table><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[# create a system user and group called hedgehog
+sudo addgroup --system hedgehog
+sudo adduser --system --ingroup hedgehog hedgehog]]></ac:plain-text-body></ac:structured-macro>
+<h2>Configure Hedgehog</h2>
+<ul>
+<li>For all installs, edit the <em>&lt;prefix&gt;/etc/hedgehog/hedgehog.yaml</em> configuration file to make sure the user names match the users specified in the previous steps.</li>
+<li>Also configure the database parameters (port, name etc...) as required.</li>
+<li>If using passwords stored in this file, be aware that this file is readable by all users.</li></ul><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[database: 
+  host         : /var/run/postgresql # specify a host for the postgresql DB. If
+                                     # this begins with a slash, it specifies the
+                                     # directory in which the socket file is 
+                                     # stored.
+  port         : 5432      # specify port for the postgresql DB.
+  name         : hedgehog  # specify dbname for the postgresql DB.
+  owner        : hedgehog  # specify a user to own the postgresql DB. 
+                           # [Required for Data Manager component]
+  owner_pass   :           # specify a password for the owner user if needed.
+  read_user    : www-data  # specify a read user for the postgresql DB. 
+                           # [Required for Web front-end component]
+  read_pass    :           # specify a password for the read user if needed.
+]]></ac:plain-text-body></ac:structured-macro><ac:structured-macro ac:name="warning"><ac:rich-text-body>
+<p><span>Do not edit the 'directories' section of this file as in 2.0 it is auto-generated.</span></p></ac:rich-text-body></ac:structured-macro>
+<h2>Data Manager</h2>
+<h3>Specify the nodes/servers</h3><ac:structured-macro ac:name="note"><ac:rich-text-body>
+<p><span>For this version of Hedgehog the servers and nodes to be processed and displayed must be specified manually.</span></p></ac:rich-text-body></ac:structured-macro>
+<ul>
+<li>Edit the the&nbsp;<em>&lt;prefix&gt;/etc/hedgehog/nodes.csv</em>&nbsp;file to specify the servers, nodes and grouping to be used (example format is provided with entries commented out).</li>
+<li>Note that the current GUI layout is optimised for nodes with short names (&lt;6 characters) of the same length</li></ul>
+<h3><span>Directory permissions</span></h3>
+<p><span>Change the ownership of the directory specified below so it is owned by the same user as the database:</span></p><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[sudo chown -R <DB_OWNER>:<DB_OWNER> <prefix>/var/hedgehog/data]]></ac:plain-text-body></ac:structured-macro><ac:structured-macro ac:name="info"><ac:rich-text-body>
+<p>You may also need to alter the permissions on this directory to allow uploads via your chosen mechanism.</p></ac:rich-text-body></ac:structured-macro>
+<p>&nbsp;</p>
+<h3>Create the database tables</h3>
+<p>Run the command below noting the following:</p>
+<ul>
+<li>If you have historic data to import then use the -m flag to specify the month of the oldest data that will need importing. Otherwise the database tables will be created to hold data from this month onwards.</li>
+<li>Note that this script will also create the directory structure for all the specified servers and nodes under the <em>data</em> directory if it does not exist</li></ul><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[sudo -u <DB_OWNER> <prefix>/bin/hedgehogctl database_init]]></ac:plain-text-body></ac:structured-macro>
+<p>&nbsp;</p>
+<h2>Web component</h2>
+<h3>Install R packages</h3>
+<p><span>These instructions</span><span>&nbsp;are for R packages that must be installed using R's built-in compilation tools (there is no Ubuntu package for them). To install interactively:</span></p><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[sudo R
+install.packages(c("brew","Cairo","googleVis","RPostgreSQL","R.utils","yaml","dplyr"))
+# you will be prompted to choose a mirror site for the repo before proceeding...
+install.packages("<prefix>/share/hedgehog/R_packages/RPostgreSQLHelper", repos=NULL)
+q()
+# If you are prompted to save workspace image y/n/c, choose no. ]]></ac:plain-text-body></ac:structured-macro>
+<p><span><span>For scripting purposes a repo can be specified by using a command of the form</span></span></p><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[install.packages("name", repos='http://cran.rstudio.com/')]]></ac:plain-text-body></ac:structured-macro>
+<p><span style="line-height: 1.4285715;">Hedgehog is tested against version 3.1.1 of R. </span>The package versions that Hedgehog has been tested against are listed below alongside each package.</p>
+<table>
+<tbody>
+<tr>
+<th>R Package</th>
+<th>Supported Version</th></tr>
+<tr>
+<td>brew</td>
+<td>1.0-6</td></tr>
+<tr>
+<td>Cairo</td>
+<td>1.5-6</td></tr>
+<tr>
+<td>googleVis</td>
+<td>0.5.8</td></tr>
+<tr>
+<td colspan="1">RPostgreSQL</td>
+<td colspan="1">0.4</td></tr>
+<tr>
+<td colspan="1">R.utils</td>
+<td colspan="1">2.0.2</td></tr>
+<tr>
+<td colspan="1">yaml</td>
+<td colspan="1">2.1.13</td></tr>
+<tr>
+<td colspan="1">dplyr</td>
+<td colspan="1">
+<p class="p1"><span class="s1">0.4.1</span></p></td></tr></tbody></table>
+<h3>Web configuration</h3>
+<p>Check the parameters in the <em>&lt;prefix&gt;/etc/hedgehog/hedgehog_gui.yaml</em> file, which specifies parameters controlling the behaviour of the web front end. See the &quot;Plot Caching&quot; section in the user guide for a more detailed description of when plots are cached.</p><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[---
+# YAML config for hedgehog GUI.
+# NOTE: If this file is changed then apache must be restarted for the changes to take effect
+www:
+  default_plot_type             : interactive # 'static'      -> png plots
+                                              # 'interactive' -> googlevis plots
+  default_interactive_plot_type : svg         # 'flash' -> plot requires flash
+                                              # 'svg'   -> plot is SVG/VML and 
+                                              # does not require flash (but with 
+                                              # svg plots some legends do not wrap 
+                                              # properly)
+  use_plot_caching              : 1           # '1' -> true, use cached plots when 
+                                              # possible 
+                                              # '0' -> false, never use cached 
+                                              # plots
+  caching_delay_in_hours        : 1           # If 'use_plot_caching=1' then only 
+                                              # plots with an end time earlier 
+                                              # than this number of hours ago are 
+                                              # cached. More recent plots are not 
+                                              # cached as data may still be being
+                                              # imported
+  presentation_delay_in_hours   : 0           # Number of hours behind now for  
+                                              # which the GUI will display data ]]></ac:plain-text-body></ac:structured-macro>
+<h3>Directory permissions</h3>
+<p>Change the ownership of the directory specified below so it is owned by the read user of the database:</p><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[sudo chown -R <DB_READ_USER>:<DB_READ_USER> <prefix>/var/hedgehog/www]]></ac:plain-text-body></ac:structured-macro>
+<p>&nbsp;</p>
+<h3>Configure apache</h3><ac:structured-macro ac:name="note"><ac:rich-text-body>
+<p>You will probably need to edit the <em>/etc/apache2/apache2.conf</em> file to enable access to the Hedgehog directories by adding <em>&lt;Directory&gt; </em>elements for <em> </em></p>
+<ul>
+<li><em>&lt;prefix&gt;/share/hedgehog&gt;</em> and<em> </em></li>
+<li><em>&lt;prefix&gt;/var/hedgehog/www&gt;&nbsp;</em></li></ul></ac:rich-text-body></ac:structured-macro>
+<p>&nbsp;</p>
+<p>Depending on your exact installation choices and apache configuration y<span style="line-height: 1.4285715;">ou may want to disable the default site using the following command:</span></p><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[sudo a2dissite 000-default.conf]]></ac:plain-text-body></ac:structured-macro>
+<p>&nbsp;</p>
+<ul>
+<li>Add the Hedgehog configuration files to apache and enable the site (this file name can be changed if required to match any local apache policy):</li></ul><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[sudo cp <prefix>/share/hedgehog/conf/hedgehog.conf /etc/apache2/sites-available/
+sudo a2ensite hedgehog.conf]]></ac:plain-text-body></ac:structured-macro><ac:structured-macro ac:name="note"><ac:rich-text-body>
+<p><span>apache/rapache write some of their logs to user.* so it can be useful to change the syslog config: </span></p><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[sudo vi /etc/rsyslog.d/50-default.conf]]></ac:plain-text-body></ac:structured-macro>
+<p>Uncomment the line&nbsp;beginning&nbsp;'user.*'.&nbsp;</p></ac:rich-text-body></ac:structured-macro>
+<p>&nbsp;</p>
+<ul>
+<li>Finally, restart apache:</li></ul><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[sudo service apache2 restart]]></ac:plain-text-body></ac:structured-macro>
+<p>&nbsp;</p>
+<p>At this point you should test that you can see the servers and nodes in the web front end at the URL http://&lt;server-name&gt;/hedgehog</p>
+<h2>Importing data</h2>
+<p>Hedgehog can process data in the following 3 ways:</p>
+<table>
+<tbody>
+<tr>
+<th>Source format</th>
+<th>Output format</th>
+<th>&nbsp;</th></tr>
+<tr>
+<td>XML</td>
+<td>Database</td>
+<td>For real time uploads</td></tr>
+<tr>
+<td colspan="1">DAT</td>
+<td colspan="1">Database</td>
+<td colspan="1">For import of historic data</td></tr>
+<tr>
+<td colspan="1">XML</td>
+<td colspan="1">DAT</td>
+<td colspan="1">For backwards compatibility with DSC</td></tr></tbody></table>
+<p>&nbsp;</p>
+<p>In each case the <em>&lt;prefix&gt;/bin/refile_and_grok</em> script is used, it is simply given different parameters:</p><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[> refile_and_grok -h
+
+refile_and_grok - finds all input files in the working directory and processes to output format
+
+-w Working directory to search for input files (default: <prefix>/var/hedgehog/data)
+-i Input file format <XML|DAT> (default: XML)
+-o Output file format <DAT|DB> (default: DB)
+-c Non-interactive mode - use this flag when being run by a cron job
+-s Start date from which to process incoming data (XML input only)
+-r Disable processing of rssac datasets. Default is to process all datasets.
+-R Reserved processors. Number of CPUS processors to exclude from import (default 0).
+-h Show this help.]]></ac:plain-text-body></ac:structured-macro>
+<p>&nbsp;</p>
+<h3>Importing historical .DAT data</h3><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[sudo -u <DB_OWNER> <prefix>/bin/refile_and_grok -i DAT]]></ac:plain-text-body></ac:structured-macro>
+<p>Be aware that this can take a long time if there is a significant amount of historic data and it may be advisable to run this in stages.</p>
+<h3>Importing real-time .XML data</h3>
+<h4>Manually</h4>
+<ul>
+<li>This can be done manually by running the <em>refile_and_grok</em> script (consider running this nohup as it may take a while depending on how much data there is to process).</li></ul><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[sudo -u <DB_OWNER> <prefix>/bin/refile_and_grok]]></ac:plain-text-body></ac:structured-macro>
+<ul>
+<li>A snapshot of the progress of the data import can be generated by running the command below:</li></ul><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[sudo -u <DB_OWNER> <prefix>/bin/hedgehogctl datafiles_create_summary]]></ac:plain-text-body></ac:structured-macro>
+<p>&nbsp;</p>
+<h4>Automatically</h4>
+<ul>
+<li>Configure a regular cron job for <em>refile_and_grok</em> as shown below</li></ul>
+<h2>Cron jobs</h2>
+<p>In 2.0 several cron jobs need to be configured.</p>
+<h3>Data Manager</h3>
+<p>Below is an example crontab for a typical data manager install (<em>sudo -u &lt;DB_OWNER&gt; crontab -e</em>).</p><ac:structured-macro ac:name="warning"><ac:rich-text-body>
+<p><strong><span>Note that the </span><em>database_manage_partitions </em><span>script </span><strong>MUST </strong> be run at least once a month to create the tables for next month or the import will fail.</strong></p></ac:rich-text-body></ac:structured-macro><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[# REQUIRED: 
+# Import XML data every 15 mins
+00,15,30,45 * * * * <prefix>/bin/refile_and_grok -c >> /home/hedgehog/refile_and_grok_xml_to_db.og 2>&1
+# Twice monthly job to make sure the DB tables for next month are created 
+# ahead of time
+0 6 15,28 * * <prefix>/bin/hedgehogctl database_manage_partitions >> /home/hedgehog/database_manage_partitions.log 2>&1
+
+# OPTIONAL:
+# Daily job to process RSSAC data. By default data is processed 
+# for a single day 1 week ago
+0 1 * * * <prefix>/bin/hedgehogctl database_process_rssac_data -D >> /home/hedgehog/database_process_rssac_data.log 2>&1
+# Monthly job to tar up processed xml directories
+0 2 1 * * <prefix>/bin/hedgehogctl datafiles_rm_empty_xml_dirs -D >> /home/hedgehog/datafiles_rm_empty_xml_dirs.log 2>&1
+# Monthly job to remove empty xml directories that are older than 7 days old
+0 2 7 * * <prefix>/bin/hedgehogctl datafiles_tar_old_xml -D >> /home/hedgehog/datafiles_tar_old_xml.log 2>&1]]></ac:plain-text-body></ac:structured-macro>
+<h3>Web component</h3>
+<p>Below is an example crontab for a typical web front-end install (<em>sudo -u &lt;DB_READ_USER&gt; crontab -e</em>)</p><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[# OPTIONAL: 
+# Daily job to create cached plots for the previous day to make loading common plots  
+# quicker. Run a few hours after midnight so all data is uploaded.
+0 4 * * * <prefix>/bin/hedgehogctl plotcache_generate_cached_plots -D >> /home/hedgehog/plotcache_generate_cached_plots.log -D 2>&1
+# Daily job to generate RSSAC reports. By default report is generated 
+# for a single day 1 week ago
+0 2 * * * <prefix>/bin/hedgehogctl rssac_generate_reports -D >> /home/hedgehog/rssac_generate_reports.log 2>&1]]></ac:plain-text-body></ac:structured-macro></ac:rich-text-body></ac:structured-macro>
\ No newline at end of file
--- /dev/null
+++ hedgehog-2.1.0~b2/docs/src/Hedgehog_Upgrade_Guide.xml
@@ -0,0 +1,224 @@
+
+<p>This document describes how to upgrade a server from release 2.0.0.b2 to release 2.0.0.rc2 of Hedgehog</p><ac:structured-macro ac:name="info"><ac:rich-text-body>
+<p>2.0.0rc1 and 2.0.0rc2 differ only in the inclusion of documentation source code to support packaging. Users already on rc1 do not need to upgrade to rc2.</p></ac:rich-text-body></ac:structured-macro>
+<p>&nbsp;</p>
+<p><ac:structured-macro ac:name="toc"><ac:parameter ac:name="style">none</ac:parameter></ac:structured-macro></p>
+<p><strong><span style="line-height: 1.5625;"><br /></span></strong></p>
+<p><strong><span style="line-height: 1.5625;">Back up the old installation</span></strong></p>
+<p><span style="line-height: 1.5625;">If desired, back up the existing installation (the directory structure in the User Guide is a useful reference) and database.&nbsp;</span></p>
+<h2><span style="line-height: 1.5625;">1. Pre-installation</span></h2>
+<h3>1.1 Changes to scripts</h3><ac:structured-macro ac:name="info"><ac:rich-text-body>
+<p>There have been several changes since the 2.0.0b2 release related to the tools.</p>
+<ul>
+<li>If you have cron jobs configured, please read the notes below and update as necessary.&nbsp;</li>
+<li>The Installation Guide provides examples of setting up the 2.0.0rc2 cron jobs for the different users/components.</li></ul></ac:rich-text-body></ac:structured-macro>
+<ol>
+<li>The Tools scripts have moved to &lt;prefix&gt;/libexec<span>/hedgehog/</span>
+<ol>
+<li>Note that refile_and_grok is still in &lt;prefix&gt;/bin</li></ol></li>
+<li>There is a wrapper provided script to run the Tools: &nbsp;&lt;prefix&gt;/bin/hedgehogctl</li>
+<li>The .sh extension has been removed from all the scripts/tools</li>
+<li>The hedgehog_ prefix has been removed from all the Tools scripts</li>
+<li>Some parameters have been changed for improved consistency</li>
+<li>There has been a change to which user should run each script, to reduce the use of the&nbsp;<span style="line-height: 1.4285715;">&lt;DB_OWNER&gt; user for split installs</span></li></ol>
+<p><span><br /></span><br /><span>The table below details the changes.</span></p>
+<table>
+<tbody>
+<tr>
+<th colspan="2" style="text-align: center;">2.0.0b2</th>
+<th colspan="2" style="text-align: center;">2.0.0rc2</th>
+<th colspan="1">&nbsp;</th>
+<th colspan="1">&nbsp;</th></tr>
+<tr>
+<th>name</th>
+<th colspan="1">
+<p>location under<br />&lt;prefix&gt;</p></th>
+<th>name</th>
+<th>
+<p><span>location </span><span>under <br /></span>&lt;prefix&gt;</p></th>
+<th>Notes</th>
+<th colspan="1">Script user</th></tr>
+<tr>
+<td>refile_and_grok.sh</td>
+<td colspan="1">bin</td>
+<td><span>refile_and_grok</span></td>
+<td><span>bin</span></td>
+<td>&nbsp;</td>
+<td colspan="1">&lt;DB_OWNER&gt;</td></tr>
+<tr>
+<td>hedgehog_database_create.sh</td>
+<td colspan="1">sbin</td>
+<td>database_create</td>
+<td>libexec/hedgehog</td>
+<td>&nbsp;</td>
+<td colspan="1">Database admin</td></tr>
+<tr>
+<td colspan="1">&nbsp;</td>
+<td colspan="1">&nbsp;</td>
+<td colspan="1">hedgehogctl</td>
+<td colspan="1"><span>bin</span></td>
+<td colspan="1">Wrapper script for the tools</td>
+<td colspan="1"><span>&lt;DB_OWNER&gt; or &lt;DB_READ_USER&gt;</span></td></tr>
+<tr>
+<td>
+<p class="p1"><span class="s1" style="color: rgb(0,51,102);"><a href="http://hedgehog_conf_read.pl"><span style="color: rgb(0,51,102);">hedgehog_conf_read.pl</span></a></span></p></td>
+<td colspan="1">bin</td>
+<td>
+<p class="p1"><span class="s1" style="color: rgb(0,51,102);"><a href="http://hedgehog_conf_read.pl"><span style="color: rgb(0,51,102);">conf_read</span></a></span></p></td>
+<td><span>libexec/hedgehog</span></td>
+<td>&nbsp;</td>
+<td colspan="1">&nbsp;</td></tr>
+<tr>
+<td>
+<p class="p1"><span class="s1">hedgehog_database_init.sh</span></p></td>
+<td colspan="1">bin</td>
+<td>database_init</td>
+<td><span>libexec/hedgehog</span></td>
+<td>&nbsp;</td>
+<td colspan="1"><span>&lt;DB_OWNER&gt;</span></td></tr>
+<tr>
+<td>hedgehog_database_manage_partitions.sh</td>
+<td colspan="1">bin</td>
+<td>database_manage_partitions</td>
+<td><span>libexec/hedgehog</span></td>
+<td>The '-p' flag has changed to '-m'</td>
+<td colspan="1"><span>&lt;DB_OWNER&gt;</span></td></tr>
+<tr>
+<td>
+<p class="p1"><span class="s2">hedgehog_database_update_nodes.sh</span></p></td>
+<td colspan="1">bin</td>
+<td>database_update_nodes</td>
+<td><span>libexec/hedgehog</span></td>
+<td>&nbsp;</td>
+<td colspan="1"><span>&lt;DB_OWNER&gt;</span></td></tr>
+<tr>
+<td>hedgehog_datafiles_create_summary.sh</td>
+<td colspan="1">bin</td>
+<td>datafiles_create_summary</td>
+<td><span>libexec/hedgehog</span></td>
+<td>&nbsp;</td>
+<td colspan="1"><span>&lt;DB_OWNER&gt;</span></td></tr>
+<tr>
+<td colspan="1">
+<p class="p1"><span class="s1">hedgehog_datafiles_rm_empty_xml_dirs.sh</span></p></td>
+<td colspan="1">bin</td>
+<td colspan="1">datafiles_rm_empty_xml_dirs</td>
+<td colspan="1"><span>libexec/hedgehog</span></td>
+<td colspan="1">&nbsp;</td>
+<td colspan="1"><span>&lt;DB_OWNER&gt;</span></td></tr>
+<tr>
+<td colspan="1">
+<p class="p1"><span class="s1">hedgehog_datafiles_tar_old_xml_files.sh</span></p></td>
+<td colspan="1">bin</td>
+<td colspan="1">datafiles_tar_old_xml_files</td>
+<td colspan="1"><span>libexec/hedgehog</span></td>
+<td colspan="1">&nbsp;</td>
+<td colspan="1"><span>&lt;DB_OWNER&gt;</span></td></tr>
+<tr>
+<td colspan="1">
+<p class="p1"><span class="s1">hedgehog_plotcache_generate_cached_plots.sh</span></p></td>
+<td colspan="1">bin</td>
+<td colspan="1">plotcache_generate_cached_plots</td>
+<td colspan="1"><span>libexec/hedgehog</span></td>
+<td colspan="1">&nbsp;</td>
+<td colspan="1"><span>&lt;DB_READ_USER&gt;</span></td></tr>
+<tr>
+<td colspan="1">
+<p class="p1"><span class="s1">hedgehog_plotcache_rm_cached_plots.sh</span></p></td>
+<td colspan="1">bin</td>
+<td colspan="1">plotcache_rm_cached_plots</td>
+<td colspan="1"><span>libexec/hedgehog</span></td>
+<td colspan="1">&nbsp;</td>
+<td colspan="1"><span>&lt;DB_READ_USER&gt;</span></td></tr>
+<tr>
+<td colspan="1">
+<p class="p1"><span class="s1">hedgehog_rssac_generate_reports</span></p></td>
+<td colspan="1">bin</td>
+<td colspan="1">rssac_generate_reports</td>
+<td colspan="1"><span>libexec/hedgehog</span></td>
+<td colspan="1">
+<ul>
+<li>The '-s' flag has been changed to '-d'</li>
+<li>The -D and -u flags have been removed</li></ul></td>
+<td colspan="1"><span>&lt;DB_READ_USER&gt;</span></td></tr>
+<tr>
+<td colspan="1">
+<p class="p1"><span class="s1">hedgehog_rssac_process_</span>unique_source_data.sh</p></td>
+<td colspan="1">bin</td>
+<td colspan="1">datafiles_process_rssac_data</td>
+<td colspan="1"><span>libexec/hedgehog</span></td>
+<td colspan="1">This script must now be run separately and before the rssac_generate_reports script.</td>
+<td colspan="1"><span>&lt;DB_OWNER&gt;</span></td></tr>
+<tr>
+<td colspan="1">
+<p class="p1"><span class="s1">hedgehog_rssac_rm_old_unique_source_data.sh</span></p></td>
+<td colspan="1">bin</td>
+<td colspan="1">datafiles_rm_old_rssac_data</td>
+<td colspan="1"><span>libexec/hedgehog</span></td>
+<td colspan="1">
+<p>The '-s' flag has been changed to '-d'</p></td>
+<td colspan="1"><span>&lt;DB_OWNER&gt;</span></td></tr>
+<tr>
+<td colspan="1">DDL_updates</td>
+<td colspan="1"><span>bin</span></td>
+<td colspan="1"><span>DDL_updates</span></td>
+<td colspan="1"><span>libexec/hedgehog</span></td>
+<td colspan="1">&nbsp;</td>
+<td colspan="1"><span>&lt;DB_OWNER&gt;</span></td></tr></tbody></table>
+<h3>1.2 Changes to directory permissions</h3><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[sudo chown -R <DB_READ_USER>:<DB_READ_USER> <prefix>/var/hedgehog/www]]></ac:plain-text-body></ac:structured-macro>
+<h3>1.3 Download, configure and make new release</h3>
+<p><span>Download the 2.0.0rc2 release from github: </span><a href="https://github.com/dns-stats/hedgehog/archive/2.0.0rc2.tar.gz">https://github.com/dns-stats/hedgehog/archive/2.0.0rc2.tar.gz</a></p>
+<p>(If you downloaded the above URL with wget the tarball will be called 2.0.0rc2.tar.gz)</p><ac:structured-macro ac:name="code"><ac:parameter ac:name="language">bash</ac:parameter><ac:plain-text-body><![CDATA[tar -xzf hedgehog-2.0.0rc2.tar.gz
+cd hedgehog-2.0.0.rc2
+./autogen.sh
+mkdir build
+cd build
+../configure
+make]]></ac:plain-text-body></ac:structured-macro>
+<h3>1.4 Cron jobs</h3>
+<p>Stop refile and grok cron job running:</p><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[sudo -u hedgegho crontab -e
+# Import XML data every 15 mins
+00,15,30,45 * * * * <prefix>/bin/refile_and_grok.sh -c >> /home/hedgehog/refile_and_grok_xml_to_db.sh.log 2>&1]]></ac:plain-text-body></ac:structured-macro>
+<p>Comment out any other jobs that are about to run. If the dsc-extractor process is running wait for it to finish before continuing.</p>
+<h3>1.5 Disable web front end</h3>
+<p>It is recommended to disable the web front end during upgrade e.g by enabling a holding page.</p>
+<h2>2. Installation</h2>
+<h3>2.1 Install new release</h3><ac:structured-macro ac:name="code"><ac:parameter ac:name="language">bash</ac:parameter><ac:plain-text-body><![CDATA[cd hedgehog-2.0.0rc2/build
+sudo make install]]></ac:plain-text-body></ac:structured-macro><ac:structured-macro ac:name="note"><ac:rich-text-body>
+<p><span>Note that in rc2 'make install-rpg' step is no longer required on install. </span></p></ac:rich-text-body></ac:structured-macro>
+<h3>2.2 Remove moved/deleted files</h3>
+<p>Some files were moved or deleted between b2 and rc2. To clean the install directory run the following commands</p><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[cd <prefix>/include
+sudo rm DSCDataManager.h DSCIOManager.h \
+        DSCStrategyFactory.h DSCStrategy.h \
+        dsc_types.h Format1DSCStrategy.h \
+        Format2DSCStrategy.h Format3aDSCStrategy.h \
+        Format3bDSCStrategy.h Format4DSCStrategy.h
+cd <prefix>/lib
+sudo rm libdsc.a libdsc.la libdsc.so libdsc.so.0 libdsc.so.0.0.0
+cd <prefix>/bin
+sudo rm -rf DDL_updates
+sudo rm hedgehog_conf_read.pl \
+        hedgehog_database_init.sh \
+        hedgehog_database_manage_partitions.sh \
+        hedgehog_database_update_nodes.sh 
+        hedgehog_datafiles_create_summary.sh \
+        hedgehog_datafiles_rm_empty_xml_dirs.sh 
+        hedgehog_datafiles_tar_old_xml_files.sh \
+        hedgehog_plotcache_generate_cached_plots.sh 
+        hedgehog_plotcache_rm_cached_plots.sh \
+        hedgehog_rssac_generate_reports.sh 
+        hedgehog_rssac_process_unique_source_data.sh \
+        hedgehog_rssac_rm_old_unique_source_data.sh refile_and_grok.sh
+sudo rm <prefix>/sbin/hedgehog_database_create.sh
+ ]]></ac:plain-text-body></ac:structured-macro>
+<h2>3. Post-install</h2>
+<h3>3.1 Test web front end</h3>
+<p>Re-enable and test the web front end. Make sure to restart apache.</p><ac:structured-macro ac:name="code"><ac:parameter ac:name="language">bash</ac:parameter><ac:plain-text-body><![CDATA[sudo service apache2 reload]]></ac:plain-text-body></ac:structured-macro>
+<h3>3.2 Re-enable cron jobs&nbsp;</h3><ac:structured-macro ac:name="warning"><ac:rich-text-body>
+<p>Remember to use the new script names and syntax when re-enabling the cron jobs</p></ac:rich-text-body></ac:structured-macro>
+<p>&nbsp;</p>
+<p>Restart refile and grok cron job running:</p><ac:structured-macro ac:name="code"><ac:plain-text-body><![CDATA[sudo -u hedgehog crontab -e
+# Import XML data every 15 mins
+00,15,30,45 * * * * <prefix>/bin/refile_and_grok -c >> /home/hedgehog/refile_and_grok_xml_to_db.log 2>&1]]></ac:plain-text-body></ac:structured-macro>
+<p>Re-enable any other jobs that were disabled for the upgrade.&nbsp;</p>
+<p>&nbsp;</p>
\ No newline at end of file
